src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        Random r = new Random();
+        Figure[] figures = {new Circle(new ColorSupplier().getRandomColor(), r.nextInt()),
+                new Square(new ColorSupplier().getRandomColor(), r.nextInt()),
+            new RightTriangle(new ColorSupplier().getRandomColor(), r.nextInt(), r.nextInt()),
+            new Rectangle(new ColorSupplier().getRandomColor(), r.nextInt(), r.nextInt()),
+            new IsoscelesTrapezoid(new ColorSupplier().getRandomColor(),
```

Comment:
colorSupplier should be `private final` class-level field
```suggestion
        Figure[] figures = {new Circle(colorSupplier.getRandomColor(), r.nextInt()),
                new Square(colorSupplier.getRandomColor(), r.nextInt()),
            new RightTriangle(colorSupplier.getRandomColor(), r.nextInt(), r.nextInt()),
            new Rectangle(colorSupplier.getRandomColor(), r.nextInt(), r.nextInt()),
            new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
```